odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -252,6 +253,7 @@ def write_serializer(item):
                 return item
         # get all fields that are not computed or x2many
         values = dict()
+        values_company_dependent = dict()
```

Comment:
Use a variable name that can be read and understood, and a `defaultdict` will make code easier.
```suggestion
        values_by_company = defaultdict(lambda: defaultdict(int))   # {company_id: {fname: value}}
```